### PR TITLE
[rq] Add RQ Integration

### LINKED ISSUE #962: Add RQ support
I was wondering if there was any work being done to support [RQ](https://github.com/rq/rq) so that job information is submitted to Datadog.  

If not, then I'm considering implementing the feature myself (but I think I might need a little guidence).

### PR DESCRIPTION
**Pull request by Kyle-Verhoog**
# Integration
This PR adds support for [rq](https://python-rq.org/).

It serves also as an example of how to write an integration in `ddtrace` (to be used alongside https://github.com/DataDog/dd-trace-py/pull/1086).

Closes #962

## Links

- Integration docs: [rq](https://python-rq.org/)
- Corp docs PR: 
- Example app: https://github.com/DataDog/trace-examples/pull/107 

## Checklist
- [x] Usage and configuration documentation added in `__init__.py`.
- [ ] [Corp docs](https://github.com/Datadog/documentation) are updated.
- [x] Entry added to release notes using [reno](https://docs.openstack.org/reno/latest/user/usage.html).
- [x] Distributed tracing propagation is implemented (if applicable).
- [x] Global configuration
  - [x] Inherits `DD_SERVICE` (if applicable).
  - [x] Environment variables are provided for config options.
  - [x] `ddtrace.config`.
- [x] Instance configuration
  - [x] Pin overriding.
  - [x] Service name override (if applicable).
- [x] Async (if applicable)
  - [x] Span parenting behaves as expected.
  - [x] Context propagation across async boundaries.
- [x] ~HTTP (if applicable)~
  - [x] ~500-level responses are tagged as errors.~
  - [x] ~Request headers specified in the config are stored.~
- [x] Tests
  - [x] Tests are provided for all of the above.
  - [x] Tests are added to CI (`.circleci/config.yml`).
  - [x] Functionality is maintained from original library.
  - [x] Patch test cases are added (see `test_django_patch.py` for an example).
  - [x] All Python versions that the library supports are tested.
  - [x] All relevant library versions (including the latest) are tested.

### CODE REVIEW

File: tests/contrib/patch.py [hunk 1]
```diff
@@ -65,18 +65,34 @@ def raise_if_no_attrs(f):
     required_attrs = [
```
Comment (Kyle-Verhoog) [inline tests/contrib/patch.py]: The changes made to this helper will be pulled into a separate PR once things are stable. This is the first integration that will be using the patch test cases (yay!!! 😄) and a few changes were necessary to get things working nicely.

File: ddtrace/contrib/rq/__init__.py [hunk 2]
```diff
@@ -0,0 +1,236 @@
...
+
+Worker Usage
+~~~~~~~~~~~~
+
+``ddtrace-run`` can be used to easily trace your workers::
+
+    DD_WORKER_SERVICE_NAME=myworker ddtrace-run rq worker
+
+
+See https://github.com/DataDog/trace-examples/tree/kyle-verhoog/rq/python/rq for a working example.
```
Comment (Kyle-Verhoog) [inline ddtrace/contrib/rq/__init__.py]: This link will be updated once the example is finalized and merged into the `trace-examples` repo

File: ddtrace/contrib/rq/__init__.py [hunk 3]
```diff
@@ -0,0 +1,236 @@
...
+Configuration
+~~~~~~~~~~~~~
+
+.. py:data:: ddtrace.config.rq['distributed_tracing_enabled']
+
+   If ``True`` the integration will connect the traces sent between the queuer and the RQ worker.
+
+   Default: ``False``
+
+.. py:data:: ddtrace.config.rq['service_name'] # Or via DD_RQ_SERVICE_NAME environment variable
```
Comment (Kyle-Verhoog) [inline ddtrace/contrib/rq/__init__.py]: Any ideas for how we can better represent the ways and precedence for configuration methods? Maybe a table?

File: ddtrace/contrib/rq/__init__.py [hunk 4]
```diff
@@ -0,0 +1,236 @@
...
+    'unpatch',
+]
+
+
+config._add('rq', dict(
+    service_name=get_env('rq', 'service_name', default='rq'),
+    worker_service_name=get_env('rq', 'worker_service_name', default='rq-worker'),
+    app='rq',
+    app_type=AppTypes.worker,
+    distributed_tracing_enabled=False,
```
Comment (Kyle-Verhoog) [inline ddtrace/contrib/rq/__init__.py]: I included distributed tracing as an option for the integration. While we don't handle this real well in the UI, I thought that there might be use cases (very small queue times) where it'd be beneficial to see the worker span in the same trace as the producer.

File: ddtrace/contrib/rq/__init__.py [hunk 5]
```diff
@@ -0,0 +1,236 @@
...
+    with pin.tracer.trace('rq.job.fetch_many', service=pin.service) as span:
+        span.set_tag('job_ids', args[0])
+        return func(*args, **kwargs)
+
+
+def patch():
+    # Avoid importing rq at the module level, eventually will be an import hook
+    import rq
+
+    if getattr(rq, '_datadog_patch', False):
```
Comment (Kyle-Verhoog) [inline ddtrace/contrib/rq/__init__.py]: This is a common pattern we have across all integrations. I wonder if there's a way we can standardize it or if it's just fine to leave it be as is. We could do a context manager or a decorator on `patch`.

Also, are we actually happy with how we do this check? It isn't thread-safe.

File: ddtrace/contrib/rq/__init__.py (hunk 3 above)
Comment (jd) [reply ddtrace/contrib/rq/__init__.py]: We should already expose a table with all those config variables :(

File: ddtrace/contrib/rq/__init__.py [hunk 6]
```diff
@@ -0,0 +1,236 @@
...
+    finally:
+        if job.get_status() == rq.job.JobStatus.FAILED:
+            span.error = 1
+            span.set_tag('exc_info', job.exc_info)
+        span.set_tag('status', job.get_status())
+        span.set_tag('origin', job.origin)
+
+        # DEV: force flush to agent since the process `os.exit()`s
+        #      immediately after this method returns
+        pin.tracer.writer.stop()
```
Comment (jd) [inline ddtrace/contrib/rq/__init__.py]: I think it's ok for now but this is a big hint that the `Tracer` API is not good enough.

File: ddtrace/contrib/rq/__init__.py [hunk 7]
```diff
@@ -0,0 +1,236 @@
...
+        pin.tracer.writer.flush_queue()
```
Comment (jd) [inline ddtrace/contrib/rq/__init__.py]: I think this is good enough, you don't need to `stop` in the first place.

File: tests/contrib/patch.py (hunk 1 above)
Comment (jd) [reply tests/contrib/patch.py]: 👍 for a separate PR

File: ddtrace/contrib/rq/__init__.py (hunk 3 above)
Comment (Kyle-Verhoog) [reply ddtrace/contrib/rq/__init__.py]: I believe we do for tracer-level environment variables but not for integration specific variables.

File: ddtrace/contrib/rq/__init__.py [hunk 8]
```diff
@@ -0,0 +1,244 @@
...
+
+Worker Usage
+~~~~~~~~~~~~
+
+``ddtrace-run`` can be used to easily trace your workers::
+
+    DD_RQ_WORKER_SERVICE=myworker ddtrace-run rq worker
+
+
+See https://github.com/DataDog/trace-examples/tree/kyle-verhoog/rq/python/rq
```
Comment (Kyle-Verhoog) [inline ddtrace/contrib/rq/__init__.py:24]: Have to change this URL out and merge the PR in trace-examples

File: ddtrace/contrib/rq/__init__.py [hunk 9]
```diff
@@ -0,0 +1,244 @@
...
+
+config._add(
+    "rq_worker",
+    dict(
+        _default_service="rq-worker",
+    ),
+)
+
+
+propagator = HTTPPropagator()
```
Comment (P403n1x87) [inline ddtrace/contrib/rq/__init__.py:91]: I think all the methods of `HTTPPropagator` are static so no need of an instance

File: ddtrace/contrib/rq/__init__.py [hunk 10]
```diff
@@ -0,0 +1,244 @@
...
+
+
+@trace_utils.with_traced_module
+def traced_queue_enqueue_job(rq, pin, func, instance, args, kwargs):
+    job = args[0]
```
Comment (P403n1x87) [inline ddtrace/contrib/rq/__init__.py:96]: just to check there is no need to check `kwargs` here

File: ddtrace/contrib/rq/__init__.py [hunk 11]
```diff
@@ -0,0 +1,244 @@
...
+
+    with pin.tracer.trace(
+        "rq.queue.enqueue_job",
+        service=trace_utils.int_service(pin, config.rq),
+        span_type=SpanTypes.WORKER,
+        resource=job.func_name,
+    ) as span:
+        span.set_tag("queue.name", instance.name)
```
Comment (P403n1x87) [inline ddtrace/contrib/rq/__init__.py:104]:
```suggestion
        span._set_str_tag("queue.name", instance.name)
```

File: ddtrace/contrib/rq/__init__.py [hunk 12]
```diff
@@ -0,0 +1,244 @@
...
+        span.set_tag("job.id", job.get_id())
+        span.set_tag("job.func_name", job.func_name)
```
Comment (P403n1x87) [inline ddtrace/contrib/rq/__init__.py:106]:
```suggestion
        span._set_str_tag("job.func_name", job.func_name)
```

File: ddtrace/contrib/rq/__init__.py [hunk 13]
```diff
@@ -0,0 +1,244 @@
...
+            # If the queue is not async then the job is run immediately so we can
+            # report the results
+            if not instance.is_async:
+                span.set_tag("job.status", job.get_status())
+
+
+@trace_utils.with_traced_module
+def traced_queue_fetch_job(rq, pin, func, instance, args, kwargs):
+    with pin.tracer.trace("rq.queue.fetch_job", service=trace_utils.int_service(pin, config.rq)) as span:
+        span.set_tag("job.id", args[0])
```
Comment (P403n1x87) [inline ddtrace/contrib/rq/__init__.py:124]: no need to check `kwargs`?

File: ddtrace/contrib/rq/__init__.py (hunk 9 above)
Comment (Kyle-Verhoog) [reply ddtrace/contrib/rq/__init__.py:91]: Yup this has changed since this PR was started 😆

File: ddtrace/contrib/rq/__init__.py (hunk 10 above)
Comment (Kyle-Verhoog) [reply ddtrace/contrib/rq/__init__.py:96]: good catch! I'll update these with the helper

File: ddtrace/contrib/rq/__init__.py (hunk 8 above)
Comment (brettlangdon) [reply ddtrace/contrib/rq/__init__.py:24]: is this still a TODO?

File: ddtrace/contrib/rq/__init__.py [hunk 14]
```diff
@@ -0,0 +1,245 @@
...
+    from ddtrace import patch
+    patch(rq=True)
+
+
+Worker Usage
+~~~~~~~~~~~~
+
+``ddtrace-run`` can be used to easily trace your workers::
+
+    DD_RQ_WORKER_SERVICE=myworker ddtrace-run rq worker
```
Comment (brettlangdon) [inline ddtrace/contrib/rq/__init__.py:21]:
```suggestion
    DD_SERVICE=myworker ddtrace-run rq worker
```

File: ddtrace/contrib/rq/__init__.py [hunk 15]
```diff
@@ -0,0 +1,245 @@
...
+
+
+See https://github.com/DataDog/trace-examples/tree/kyle-verhoog/rq/python/rq
+for a working example.
+
+
+Instance Configuration
+~~~~~~~~~~~~~~~~~~~~~~
```
Comment (brettlangdon) [inline ddtrace/contrib/rq/__init__.py:29]: TODO?

File: ddtrace/contrib/rq/__init__.py [hunk 16]
```diff
@@ -0,0 +1,245 @@
...
+
+
+
+Global Configuration
+~~~~~~~~~~~~~~~~~~~~
+
+.. py:data:: ddtrace.config.rq['distributed_tracing_enabled']
+
+   If ``True`` the integration will connect the traces sent between the queuer
```
Comment (brettlangdon) [inline ddtrace/contrib/rq/__init__.py:38]:
```suggestion
   If ``True`` the integration will connect the traces sent between the producer
```

? is `queuer` a term used by RQ?

File: ddtrace/contrib/rq/__init__.py (hunk 14 above)
Comment (brettlangdon) [reply ddtrace/contrib/rq/__init__.py:21]: hmmm, I wonder if we should get rid of the worker separate service name and just keep as a single service name for the integration which defaults to `DD_SERVICE` or maybe `rq` if unset ?

File: ddtrace/contrib/rq/__init__.py [hunk 17]
```diff
@@ -0,0 +1,245 @@
...
+    """Trace rq.Job.fetch_many(...)"""
+    with pin.tracer.trace("rq.job.fetch_many", service=trace_utils.ext_service(pin, config.rq_worker)) as span:
+        job_ids = get_argument_value(args, kwargs, 0, "job_ids")
+        span.set_tag("job_ids", job_ids)
+        return func(*args, **kwargs)
+
+
+def patch():
+    # Avoid importing rq at the module level, eventually will be an import hook
+    import rq
```
Comment (brettlangdon) [inline ddtrace/contrib/rq/__init__.py:199]: so with this approach we'd end up with an import error on calling `patch` rather than an attribute error from `patch` no existing?

File: ddtrace/contrib/rq/__init__.py [hunk 18]
```diff
@@ -0,0 +1,245 @@
...
+        if instance.is_async and config.rq.distributed_tracing_enabled:
+            HTTPPropagator.inject(span.context, job.meta)
+
+        try:
+            return func(*args, **kwargs)
+        finally:
+            # If the queue is not async then the job is run immediately so we can
+            # report the results
+            if not instance.is_async:
+                span.set_tag("job.status", job.get_status())
```
Comment (brettlangdon) [inline ddtrace/contrib/rq/__init__.py:116]: do we also need to set `span.error` ?

File: ddtrace/contrib/rq/__init__.py [hunk 19]
```diff
@@ -0,0 +1,245 @@
...
+    try:
+        with pin.tracer.trace("rq.job.fetch", service=trace_utils.ext_service(pin, config.rq_worker)) as span:
+            job_id = get_argument_value(args, kwargs, 0, "id")
+            span.set_tag("job.id", job_id)
+            job = func(*args, **kwargs)
+            return job
+    finally:
+        if job:
+            job_status = job.get_status()
+            span.set_tag("job.status", job_status)
```
Comment (brettlangdon) [inline ddtrace/contrib/rq/__init__.py:185]: `span.error`?

File: riotfile.py [hunk 20]
```diff
@@ -1012,6 +1012,42 @@ def select_pys(min_version=MIN_PYTHON_VERSION, max_version=MAX_PYTHON_VERSION):
...
+                            "~=1.0.0",
+                            "~=1.1.0",
+                            "~=1.2.0",
+                            "~=1.3.0",
+                        ],
+                        "redis": ["==3.0"],
+                    },
+                ),
+                Venv(
+                    pys=["3.5", "3.6", "3.7", "3.8", "3.9"],
```
Comment (brettlangdon) [inline riotfile.py:1032]: `select_pys`?

File: riotfile.py [hunk 21]
```diff
@@ -1012,6 +1012,42 @@ def select_pys(min_version=MIN_PYTHON_VERSION, max_version=MAX_PYTHON_VERSION):
                 ),
             ],
         ),
+        Venv(
+            name="rq",
+            command="pytest tests/contrib/rq",
+            venvs=[
+                Venv(
+                    pys=["2.7"],
```
Comment (brettlangdon) [inline riotfile.py:1020]: `select_pys`?

File: riotfile.py [hunk 22]
```diff
@@ -1012,6 +1012,42 @@ def select_pys(min_version=MIN_PYTHON_VERSION, max_version=MAX_PYTHON_VERSION):
...
+                            "~=1.0.0",
+                            "~=1.1.0",
+                            "~=1.2.0",
+                            "~=1.3.0",
+                            "~=1.4.0",
+                            "~=1.5.0",
+                            "~=1.6.0",
+                            "~=1.7.0",
+                            "~=1.8.0",
+                            "~=1.9.0",
```
Comment (brettlangdon) [inline riotfile.py:1044]: latest?

File: tests/contrib/rq/test_rq.py [hunk 23]
```diff
@@ -0,0 +1,116 @@
...
+    patch()
+    try:
+        sync_q = rq.Queue("sync-q", is_async=False, connection=connection)
+        yield sync_q
+    finally:
+        unpatch()
+
+
+@snapshot(ignores=snapshot_ignores)
+def test_queue_enqueue(sync_queue):
```
Comment (brettlangdon) [inline tests/contrib/rq/test_rq.py:50]:
```suggestion
def test_sync_queue_enqueue(sync_queue):
```

?

File: tests/snapshots/tests.contrib.rq.test_rq.test_enqueue.snap [hunk 24]
```diff
@@ -0,0 +1,59 @@
+[[{"name" "rq.queue.enqueue_job"
```
Comment (brettlangdon) [inline tests/snapshots/tests.contrib.rq.test_rq.test_enqueue.snap:1]: is this trace correct? we have 3 separate traces which are not connected at all?

File: tests/snapshots/tests.contrib.rq.test_rq.test_sync_worker.snap [hunk 25]
```diff
@@ -0,0 +1,36 @@
+[[{"name" "rq.queue.enqueue_job"
```
Comment (brettlangdon) [inline tests/snapshots/tests.contrib.rq.test_rq.test_sync_worker.snap:1]: same here, I wouldn't have expected 2 traces ?

File: ddtrace/contrib/rq/__init__.py (hunk 8 above)
Comment (majorgreys) [reply ddtrace/contrib/rq/__init__.py:24]: We don't often refer to trace-examples in our documentation so I would be fine with removing here.

File: ddtrace/contrib/rq/__init__.py (hunk 16 above)
Comment (majorgreys) [reply ddtrace/contrib/rq/__init__.py:38]: The RQ docs refer to "queues" and "workers". The app does the "enqueueing". So we could refer to the "enqueuer" and the "worker".

File: ddtrace/contrib/rq/__init__.py [hunk 26]
```diff
@@ -0,0 +1,245 @@
...
+
+@trace_utils.with_traced_module
+def traced_queue_enqueue_job(rq, pin, func, instance, args, kwargs):
+    job = get_argument_value(args, kwargs, 0, "f")
+
+    with pin.tracer.trace(
+        "rq.queue.enqueue_job",
+        service=trace_utils.int_service(pin, config.rq),
+        span_type=SpanTypes.WORKER,
+        resource=job.func_name,
```
Comment (majorgreys) [inline ddtrace/contrib/rq/__init__.py:100]: Looking at the RQ dashboard the monitoring users perform make reference to queue names and worker names.

I'm inclined to think that both these should show up on a span and not just as tags.

```suggestion
        resource="{} {}".format(instance.name, job.func_name),
```

File: ddtrace/contrib/rq/__init__.py [hunk 27]
```diff
@@ -0,0 +1,245 @@
...
+    # `perform_job` is executed in a freshly forked, short-lived instance
+    job = get_argument_value(args, kwargs, 0, "job")
+
+    ctx = HTTPPropagator.extract(job.meta)
+    if ctx.trace_id:
+        pin.tracer.context_provider.activate(ctx)
+
+    try:
+        with pin.tracer.trace(
+            "rq.worker.perform_job",
```
Comment (majorgreys) [inline ddtrace/contrib/rq/__init__.py:139]: This operation name seems to be redundant if we also have a service name for the worker that is different from the enqueueing app.

It seems to me that we could reduce this by using `rq.queue` and `rq.worker` for the service names and remove them from the namespace of the operation names.

The job perform and fetch would then be under the `rq.worker` (though maybe I'm getting the flow wrong).

File: tests/snapshots/tests.contrib.rq.test_rq.test_enqueue.snap [hunk 28]
```diff
@@ -0,0 +1,59 @@
...
+       "duration" 4910000
+       "meta" {"job.id" "6fcf262c-a9ff-4056-bd48-0bbca86151f7"}}]
+ [{"name" "rq.job.fetch"
+   "service" "rq-worker"
+   "resource" "rq.job.fetch"
+   "error" 0
+   "span_id" 0
+   "trace_id" 2
+   "parent_id" nil
+   "start" 1629316605892859000
```
Comment (majorgreys) [inline tests/snapshots/tests.contrib.rq.test_rq.test_enqueue.snap:52]: This start time is before `rq.worker.perform_job` which makes sense but I would have expected a span that was the parent of both here as they should be done by the worker instance in sequence.

Looking at the source code, it appears that a job is fetched and then handed off for execution within a work loop.

https://github.com/rq/rq/blob/b80045d615d539cf4c565c5541694efe13ea648e/rq/worker.py#L599
https://github.com/rq/rq/blob/b80045d615d539cf4c565c5541694efe13ea648e/rq/worker.py#L607

Being that this work is connected, these two spans should share a common parent. A possibility here could be to have a parent `work` span where the first child is the fetch. The parent span can be left open after the fetch is complete, stashed in the job object, and then closed once the job has executed.

**Review by majorgreys — COMMENTED** [review]
I had a few questions about the span data from this integration.

File: ddtrace/contrib/rq/__init__.py [hunk 29]
```diff
@@ -0,0 +1,232 @@
...
+   environment variables.
+
+   Default: ``rq``
+
+.. py:data:: ddtrace.config.rq_worker['service']
+
+   The service name reported by default for RQ spans from workers.
+
+   This option can also be set with the ``DD_RQ_WORKER_SERVICE`` environment
+   variable.
```
Comment (brettlangdon) [inline ddtrace/contrib/rq/__init__.py:70]:
```suggestion
   This option can also be set with the ``DD_RQ_WORKER_SERVICE`` or ``DD_SERVICE``
   environment variables.
```

?

File: ddtrace/contrib/rq/__init__.py (hunk 27 above)
Comment (brettlangdon) [reply ddtrace/contrib/rq/__init__.py:145]: however, this doesn't account for people having their own custom service names, keeping `rq.` in the operation name is good unless we want to start using `component` (or equivalent) tag.

File: ddtrace/contrib/rq/__init__.py [hunk 30]
```diff
@@ -0,0 +1,232 @@
...
+
+    try:
+        with pin.tracer.trace(
+            "rq.worker.perform_job",
+            service=trace_utils.int_service(pin, config.rq_worker),
+            span_type=SpanTypes.WORKER,
+            resource=job.func_name,
+        ) as span:
+            span.set_tag("job.id", job.get_id())
+            r = func(*args, **kwargs)
```
Comment (brettlangdon) [inline ddtrace/contrib/rq/__init__.py:151]: do we need a try/finally around this?

can we get `job.stats`/`job.is_failed`/`job.origin` if this raises an exception?

File: ddtrace/contrib/rq/__init__.py (hunk 29 above)
Comment (Kyle-Verhoog) [reply ddtrace/contrib/rq/__init__.py:70]: yes good catch. We should list DD_SERVICE first too

Comment (brettlangdon) [reply ddtrace/contrib/rq/__init__.py:70]: `DD_SERVICE` first would be a nice touch.

File: ddtrace/contrib/rq/__init__.py (hunk 27 above)
Comment (Kyle-Verhoog) [reply ddtrace/contrib/rq/__init__.py:145]: it also more closely matches the rq api which helps I think from a user perspective knowing what this span is wrapping

File: ddtrace/contrib/rq/__init__.py (hunk 30 above)
Comment (Kyle-Verhoog) [reply ddtrace/contrib/rq/__init__.py:151]: Ah no we can't because the trace is finished in the `with` so setting the tags afterwards is not possible!

File: ddtrace/contrib/rq/__init__.py (hunk 26 above)
Comment (Kyle-Verhoog) [reply ddtrace/contrib/rq/__init__.py:113]: Yeah so rq does this by default for functional jobs but doesn't fully qualify the classes:

https://github.com/rq/rq/blob/c556106a38505ea7485daec9420c408f720d2961/rq/job.py#L105-L117

so I'll add support for when there's an `instance`

File: ddtrace/contrib/rq/__init__.py (hunk 30 above)
Comment (brettlangdon) [reply ddtrace/contrib/rq/__init__.py:151]: I mean doing this: double up the try/finally

```suggestion
            try:
                r = func(*args, **kwargs)
            finally:
                # add the tags here
```

File: tests/snapshots/tests.contrib.rq.test_rq.test_enqueue.snap (hunk 28 above)
Comment (Kyle-Verhoog) [reply tests/snapshots/tests.contrib.rq.test_rq.test_enqueue.snap]: would there be any value in measuring the batches of jobs on the worker side? It would expose the usage patterns of the worker maybe but I'm not sure if that's useful to a user.

File: tests/snapshots/tests.contrib.rq.test_rq.test_enqueue_distributed_tracing_enabled_None_worker_service_custom-worker-service.snap [hunk 31]
```diff
@@ -0,0 +1,45 @@
...
+       "service" "custom-worker-service"
+       "resource" "tests.contrib.rq.jobs.job_add1"
+       "type" "worker"
+       "error" 0
+       "span_id" 1
+       "trace_id" 0
+       "parent_id" 0
+       "start" 1631053636948434900
+       "duration" 63030800
+       "meta" {"runtime-id" "1a0fb63e68d94c12a364d691b87726c5"
```
Comment (brettlangdon) [inline tests/snapshots/tests.contrib.rq.test_rq.test_enqueue_distributed_tracing_enabled_None_worker_service_custom-worker-service.snap:29]: we ignore these when asserting right? might be interesting if we could assert that these came from two different processes (runtime-id and system.pid) since that is what we expect.

aside/musing, not 100% related to this PR

**Review by brettlangdon — DISMISSED** [review]

**Comment by brettlangdon** [comment]
We are also missing linking RQ docs into the main integration docs page: https://830111-61572326-gh.circle-artifacts.com/0/tmp/docs/integrations.html#rq

**Review by brettlangdon — APPROVED** [review]

**Review by P403n1x87 — APPROVED** [review]
